cabal-install/src/Distribution/Client/CmdRepl.hs

```diff
@@ -554,6 +549,68 @@ requireCabal version source =
   , source
   )
 
+reportProjectNoTarget :: Flag FilePath -> [String] -> Doc
+reportProjectNoTarget projectFile pkgs =
+  case (null pkgs, projectName) of
+    (True, Just project) ->
+      text "There are no packages in"
+        <+> (project <> char '.')
+        <+> text "Please add a package to the project and"
+        <+> pickComponent
+    (True, Nothing) ->
+      text "Please add a package to the project and" <+> pickComponent
+    (False, Just project) ->
+      text "Please"
+        <+> pickComponent
+        <+> text "The packages in"
+        <+> project
+        <+> (text "from which to select a component target are" <> colon)
+        $+$ nest 1 (vcat [text "-" <+> text pkg | pkg <- sort pkgs])
+    (False, Nothing) ->
+      text "Please"
+        <+> pickComponent
+        <+> (text "The packages from which to select a component in 'cabal.project'" <> comma)
+        <+> (text "the implicit default as if `--project-file=cabal.project` was added as a command option" <> comma)
+        <+> (text "are" <> colon)
+        $+$ nest 1 (vcat [text "-" <+> text pkg | pkg <- sort pkgs])
+  where
+    projectName = case projectFile of
+      Flag "" -> Nothing
+      Flag n -> Just $ quotes (text n)
+      _ -> Nothing
+    pickComponent = text "pick a single [package:][ctype:]component as target for the REPL command."
```

Comment:
With the multi-repl you can select multiple components or "all", so this message seems a little imprecise.

Reply:
That failure was due to the two packages in directories `one` and `two` being `pkg-one`. Fixed now.

Reply:
I've added an `(or all)` to that message:

```diff
-    "pick a single [package:][ctype:]component as target for the REPL command."
+    "pick a single [package:][ctype:]component (or all) as target for the REPL command."
```

Reply:
I didn't go so far as to say that `--enable-multi-repl` would also be needed if `all` was chosen.